hyperspy/_signals/dielectric_function.py

```diff
@@ -20,15 +20,19 @@
 from scipy import constants
 from scipy.integrate import simps, cumtrapz
 
-from hyperspy.signals import Signal1D
+from hyperspy._signals.signal1d import Signal1D
+from hyperspy._signals.complex_signal import ComplexSignal
 from hyperspy.misc.eels.tools import eels_constant
 
 
-class DielectricFunction(Signal1D):
+class DielectricFunction(ComplexSignal):
+
+    _record_by = "spectrum"
     _signal_type = "DielectricFunction"
 
-    def __init__(self, *args, **kwards):
-        Signal1D.__init__(self, *args, **kwards)
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.axes_manager.set_signal_dimension(1)
```

Comment:
If we go for a `ComplexSignal1D` subclass, it'll be there were this line should go.